assets/performanceprofile/scripts/set-rps-mask.sh

```diff
@@ -1,38 +1,18 @@
 #!/usr/bin/env bash
 
-dev=$1
-[ -n "${dev}" ] || { echo "The device argument is missing" >&2 ; exit 1; }
+full_path=${1}
+[ -n "${full_path}" ] || { echo "The full device path argument is missing" >&2 ; exit 1; }
 
-mask=$2
+mask=${2}
 [ -n "${mask}" ] || { echo "The mask argument is missing" >&2 ; exit 1; }
 
-dev_dir="/sys/class/net/${dev}"
+# replace '-' with '/'
+de_escape_path="/sys${full_path//-//}"
 
-function find_dev_dir {
-  systemd_devs=$(systemctl list-units -t device | grep sys-subsystem-net-devices | cut -d' ' -f1)
+# get the path for the queues
+queues_path=${de_escape_path%/rx*}
```

Comment:
... so here you should be better off not trimming the trailing part and avoiding the loop below, with something alike:

queue_path=${de_escape_path//queues\/rx\//queues\/rx-}
echo "${mask}" > "${queue_path}"

Side question: why is this script invoked via a systemd unit and not directly by the udev rule?

Reply:
Additionally, is devpath escaping really needed here? Otherwise you can replace '%i' with '%I' in the unit file and avoid the de-escaping above.

Reply:
> why is this script invoked via a systemd unit and not directly by the udev rule?

I think the reason is because then we can use it as `oneshot` service but maybe @MarSik knows better

Reply:
The loop was just to make sure we're on the safe side and nothing get missed, but yes removing the loop should be suffice as well